### PR TITLE
should allow whitespaces in `data.table(key=)`

### LINKED ISSUE #4292: data.table(key=) doesn't allow whitespaces
## OK

```r
data.table(A=1,B=2,C=3, key="A,B")
```

## This fails

```r
data.table(A=1,B=2,C=3, key="A, B")
#  Error in setkeyv(ans, key) : some columns are not in the data.table:  B 
```

But I think it should work.

### PR DESCRIPTION
**Pull request by shrektan**
Closes #4292 

Should allow whitespaces in `data.table(key=)` when `key` is a scalar string. 

Now this works.

```r
data.table(A=1,B=2,C=3, key="A, B")
```

### CODE REVIEW

File: R/data.table.R [hunk 1]
```diff
@@ -62,7 +62,7 @@ data.table = function(..., keep.rownames=FALSE, check.names=FALSE, key=NULL, str
   if (!is.null(key)) {
     if (!is.character(key)) stop("key argument of data.table() must be character")
     if (length(key)==1L) {
-      key = strsplit(key,split=",")[[1L]]
+      key = trimws(strsplit(key,split=",")[[1L]])
```
Comment (MichaelChirico) [inline R/data.table.R]: IIRC `trimws` wasn't in R 3.1.0

Comment (MichaelChirico) [reply R/data.table.R]: hmm i may be wrong

https://github.com/wch/r-source/commit/e4f12c04ea2a0c9ff7f3fbfa445ae2ae7300200a

Comment (MichaelChirico) [reply R/data.table.R]: yep it's 3.2.0

https://github.com/wch/r-source/blob/trunk/doc/NEWS.Rd#L5382

Comment (MichaelChirico) [reply R/data.table.R]: ah right, but we back-ported it 

https://github.com/Rdatatable/data.table/blob/master/R/data.table.R#L1

**Comment by MichaelChirico** [comment]
I'm not a big fan of using `key` like this in the first place, but this extension seems relatively harmless anyway

File: R/data.table.R (hunk 1 above)
Comment (MichaelChirico) [inline R/data.table.R]: why not split on `,\\s*`?

please change to `fixed=TRUE` if we are going to keep `split=','`

Comment (shrektan) [reply R/data.table.R]: because that would not work for `key = " A,B "`, despite it's trivial.

Sure, added.

Comment (MichaelChirico) [reply R/data.table.R]: `strsplit(trimws(key), split = ',\\s*')` works though 🙃 

i'm fine with either this or the `fixed=TRUE` approach, anyway

**Comment by jangorecki** [comment]
note that I created #4357

**Comment by MichaelChirico** [comment]
@shrektan would you still like to merge this? if so I'll add it to the 1.16.0 milestone

**Comment by MichaelChirico** [comment]
Oh, just read #4357. So let's decide on deprecation first. Commented there on the need first to see how costly it will be to deprecate this approach. Marking this PR as draft for now pending that.

**Comment by jangorecki** [comment]
I think we can close this PR in favor of removing support for such strings.

**Pull request closed without merging**